frontend/packages/ux-editor/src/utils/component.ts

```diff
@@ -116,37 +189,24 @@ export const setComponentProperty = <
   [propertyKey]: value,
 });
 
-export const ExpressionSchemaBooleanDefinitionReference =
-  'expression.schema.v1.json#/definitions/boolean';
+export const EXPRESSION_SCHEMA_BASE_DEFINITION_REFERENCE =
+  'expression.schema.v1.json#/definitions/' as const;
 
-/**
- * Gets an array of unsupported property keys
- * @param properties The properties object to check.
- * @param knownUnsupportedPropertyKeys An array of additional known unsupported property keys.
- * @returns An array of unsupported property keys.
- */
-export const getUnsupportedPropertyTypes = (
-  properties: KeyValuePairs,
-  knownUnsupportedPropertyKeys?: string[],
-) => {
-  const propertyKeys = Object.keys(properties);
-  let unsupportedPropertyKeys = propertyKeys
-    .filter((key) =>
-      knownUnsupportedPropertyKeys ? !knownUnsupportedPropertyKeys.includes(key) : true,
-    )
-    .filter((key) => {
-      return !isPropertyTypeSupported(properties[key]);
-    });
-
-  if (knownUnsupportedPropertyKeys) {
-    unsupportedPropertyKeys = unsupportedPropertyKeys.concat(knownUnsupportedPropertyKeys);
-  }
-
-  return unsupportedPropertyKeys;
+export const getExpressionSchemaDefinitionReference = (type: PropertyTypes) => {
+  return `${EXPRESSION_SCHEMA_BASE_DEFINITION_REFERENCE}${type}`;
 };
 
-const supportedPropertyTypes = ['boolean', 'number', 'integer', 'string'];
-const supportedPropertyRefs = [ExpressionSchemaBooleanDefinitionReference];
+const supportedPropertyTypes = [
+  PropertyTypes.boolean,
+  PropertyTypes.number,
+  PropertyTypes.integer,
+  PropertyTypes.string,
+  PropertyTypes.object,
+  PropertyTypes.array,
+];
+const supportedPropertyRefs = supportedPropertyTypes
+  .filter((p) => p !== 'object' && p !== 'array')
```

Comment:
Can we use the enum her also?
```suggestion
  .filter((p) => p !== PropertyTypes.object && p !== PropertyTypes.array)
```